Daily executive report

trink opened the pull request.

trink committed:
Refactor the executive report plugin

- consolidate the weekly and monthly reports
- add a daily report

trink committed:
Pre-aggregrate the report on the C side and update the Lua table once

- report generation is almost 100x faster on our 150K nightly user test set
- only requires an extra 320KiB

mreid-moz commented on heka/sandbox/filters/firefox_executive_report.lua:
```diff
@@ -0,0 +1,237 @@
+        fx_cids:report(intervals[current_interval % WEEKS + 1])
+        clear_intervals(current_interval, week, WEEKS)
+    elseif delta >= WEEKS then
+        error(string.format("data gap over %d weeks", WEEKS))
```
Do we not need the check + error for delta < 0 anymore?

mreid-moz replied:
Never mind this either - the equivalent check is done in `process_message` before this is called.

mreid-moz commented on heka/sandbox/filters/firefox_executive_report.lua:
```diff
@@ -0,0 +1,237 @@
+    local interval = intervals[week % WEEKS + 1]
+    local r, country, channel, _os = get_row(interval, "%Y-%m-%d", week * SEC_IN_WEEK - (DAY_OFFSET * SEC_IN_DAY))
+    if r then
+        -- The use of submission date changes the meaning of the day of the week
```
Add a date to this comment for future archeologists, and a description of what the behaviour was before (using activity date)

mreid-moz commented on heka/sandbox/filters/firefox_executive_report.lua:
```diff
@@ -0,0 +1,237 @@
+end
+
+
+local function update_month(ts, cid, day, day_changed)
```
`day` parameter is not used

mreid-moz replied:
Never mind... I see it's for compatibility with other "update_interval" functions. Might be worth a comment saying it's not expected to be used.

trink committed:
Review feedback

- clarify that some of the parameters in the various update_interval
  functions are not used
- update the submission vs activity date reporting comment
